cilium/cmd/service_update.go

```diff
@@ -64,6 +65,53 @@ func parseFrontendAddress(address string) (*models.FrontendAddress, net.IP) {
 	}, frontend.IP
 }
 
+func appendService(svc *models.Service, fa *models.FrontendAddress, faIP net.IP, ips []net.IP, port string) error {
+	p := uint64(0)
+
+	if port != "" {
+		var err error
+		p, err = strconv.ParseUint(port, 10, 16)
+		if err != nil {
+			return fmt.Errorf("invalid port %s: %s", port, err)
+		}
+	}
+
+	for _, ip := range ips {
+		be, err := types.NewL3n4Addr(types.TCP, ip, uint16(p))
+		if err != nil {
+			return fmt.Errorf("unable to create backend address: %s", err)
+		}
+
+		if be.IsIPv6() && faIP.To4() != nil {
+			return fmt.Errorf("address mismatch between frontend and backend")
+		}
+
+		if fa.Port == 0 && p != 0 {
+			return fmt.Errorf("L4 backend found with L3 frontend")
+		}
+
+		svc.BackendAddresses = append(svc.BackendAddresses, be.GetBackendModel())
+	}
+
+	return nil
+}
+
+func parseBackend(backend string, svc *models.Service, fa *models.FrontendAddress, faIP net.IP) error {
+	if host, port, err := net.SplitHostPort(backend); err == nil {
```

Comment:
This may need to change. With the weights we specify [::1]:90/100 where 100 is the weight.